sdk/devcenter/developer-devcenter-rest/src/isUnexpected.ts

```diff
@@ -0,0 +1,460 @@
+    | EnvironmentsListCatalogItemVersions200Response
+    | EnvironmentsListCatalogItemVersionsdefaultResponse
+    | EnvironmentsGetCatalogItemVersion200Response
+    | EnvironmentsGetCatalogItemVersiondefaultResponse
+    | EnvironmentsListEnvironmentTypes200Response
+    | EnvironmentsListEnvironmentTypesdefaultResponse
+): response is
+  | DevCenterListProjectsdefaultResponse
+  | DevCenterGetProjectdefaultResponse
+  | DevCenterListAllDevBoxesdefaultResponse
+  | DevCenterListAllDevBoxesByUserdefaultResponse
+  | DevBoxesListPoolsdefaultResponse
+  | DevBoxesGetPooldefaultResponse
+  | DevBoxesListSchedulesByPooldefaultResponse
+  | DevBoxesGetScheduleByPooldefaultResponse
+  | DevBoxesListDevBoxesByUserdefaultResponse
+  | DevBoxesGetDevBoxByUserdefaultResponse
+  | DevBoxesCreateDevBoxdefaultResponse
+  | DevBoxesDeleteDevBoxdefaultResponse
+  | DevBoxesStartDevBoxdefaultResponse
+  | DevBoxesStopDevBoxdefaultResponse
+  | DevBoxesGetRemoteConnectiondefaultResponse
+  | EnvironmentsListEnvironmentsdefaultResponse
+  | EnvironmentsListEnvironmentsByUserdefaultResponse
+  | EnvironmentsGetEnvironmentByUserdefaultResponse
+  | EnvironmentsCreateOrUpdateEnvironmentdefaultResponse
+  | EnvironmentsUpdateEnvironmentdefaultResponse
+  | EnvironmentsDeleteEnvironmentdefaultResponse
+  | EnvironmentsDeployEnvironmentActiondefaultResponse
+  | EnvironmentsDeleteEnvironmentActiondefaultResponse
+  | EnvironmentsCustomEnvironmentActiondefaultResponse
+  | EnvironmentsListArtifactsByEnvironmentdefaultResponse
+  | EnvironmentsListArtifactsByEnvironmentAndPathdefaultResponse
+  | EnvironmentsListCatalogItemsdefaultResponse
+  | EnvironmentsGetCatalogItemdefaultResponse
+  | EnvironmentsListCatalogItemVersionsdefaultResponse
+  | EnvironmentsGetCatalogItemVersiondefaultResponse
+  | EnvironmentsListEnvironmentTypesdefaultResponse {
+  const lroOriginal = response.headers["x-ms-original-url"];
+  const url = new URL(lroOriginal ?? response.request.url);
+  const method = response.request.method;
+  let pathDetails = responseMap[`${method} ${url.pathname}`];
+  if (!pathDetails) {
+    pathDetails = geParametrizedPathSuccess(method, url.pathname);
+  }
+  return !pathDetails.includes(response.status);
+}
+
+function geParametrizedPathSuccess(method: string, path: string): string[] {
+  const pathParts = path.split("/");
+
+  // Iterate the responseMap to find a match
+  for (const [key, value] of Object.entries(responseMap)) {
+    // Extracting the path from the map key which is in format
+    // GET /path/foo
+    const candidatePath = getPathFromMapKey(key);
+    const candidateMethod = getMethodFromMapKey(key);
+    // Get each part of the url path
+    const candidateParts = candidatePath.split("/");
+
+    // If the candidate and actual paths don't match in size
+    // we move on to the next candidate path
+    if (
+      candidateParts.length === pathParts.length &&
+      hasParametrizedPath(key) &&
+      method == candidateMethod
```

Comment:
```suggestion
      method === candidateMethod
```

To fix ci error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1925208&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=c9190431-7417-53a5-3fc1-7c7e026cd79d&l=92.